ls1intum/Artemis
# [Enhancement] Improve various German translations

### Motivation and Context
While using Artemis in German I noticed various typos and additional spaces in compound-nouns. When I found inconsistencies (e.g. ‘Multiple Choice‘/‘Multiple-Choice’, ‘Drag And Drop Exercise’/‘Drag-And-Drop Exercise’, ‘Buildplan’/‘Build-Plan’) I chose one format.

This is most probably not an exhaustive fix, so:
If you find any more fixable translations, feel free to add your commits to this PR or add a comment below.

### Description
In general, I tried to apply these rules when fixing compound-nouns:
* If one part is German the other English (and also not common as loanword), I combined them using a dash. E.g. ‘Build-Plan’, ‘Studierenden-Repository’.
* If all parts are German and they can be combined into a proper noun, I did so without dashes. E.g. ‘Kurs Verwaltung‘ → ‘Kursverwaltung’, ‘Bewertung Kriterium’ → ‘Bewertungskriterium’.
* If this resulted in a weird combination, I used a mix. E.g. ‘Beispiel Lösungs Editor’ → ’Beispiellösungs-Editor’.

### Steps for Testing
Set the language to German. Click through the various pages in Artemis to check if a translation looks out-of-place or if you can spot additional ones that should be changed.

### Test Coverage
n/a

### Screenshots
n/a
